cosmo_tester/test_suites/manager_of_managers/regional_cluster.py

```diff
@@ -0,0 +1,974 @@
+                    },
+                ],
+
+                'deployments': [
+                    {
+                        'deployment_id': constants.HELLO_WORLD_DEP,
+                        'blueprint_id': constants.HELLO_WORLD_BP,
+                        'tenant': constants.TENANT_1,
+                        'inputs': {
+                            'region': 'RegionOne',
+                            'external_network_id':
+                                self.attributes.floating_network_id,
+                            'image': self.attributes.default_linux_image_id,
+                            'flavor': self.attributes.small_flavor_name
+                        }
+                    }
+                ]
+            }
+
+    @property
+    def network_inputs(self):
+        raise NotImplementedError('Each Regional Cluster class needs to '
+                                  'add additional network inputs')
+
+    def validate(self):
+        raise NotImplementedError('Each Regional Cluster class needs to '
+                                  'implement the `validate` method')
+
+    @property
+    def first_deployment(self):
+        """
+        Indicate that this is the initial deployment, as opposed to the second
+        one, to which we will upgrade
+        """
+        return constants.FIRST_DEP_INDICATOR in self.deployment_id
+
+    def upload_blueprint(self, use_cfy=False):
+        self.clone_example()
+        namespace_blueprint_file = self._cloned_to / \
+            'blueprints/infrastructure-examples/openstack/' \
+            'floating-ip-configuration.yaml'
+        namespace_blueprint_id = 'infra'
+        blueprint_file = self._cloned_to / self.blueprint_file
+
+        self.logger.info('Uploading blueprint: %s [id=%s]',
+                         namespace_blueprint_file,
+                         namespace_blueprint_id)
+        self.logger.info('Uploading blueprint: %s [id=%s]',
+                         blueprint_file,
+                         self.blueprint_id)
+
+        if use_cfy:
+            self.cfy.profile.set([
+                '-t', self.tenant,
+            ])
+            self.cfy.blueprint.upload([
+                '-b', namespace_blueprint_id,
+                namespace_blueprint_file
+            ])
+            self.cfy.blueprint.upload([
+                '-b', self.blueprint_id,
+                blueprint_file
+            ])
+        else:
+            if self.first_deployment:
+                with util.set_client_tenant(self.manager, self.tenant):
+                    self.manager.client.blueprints.upload(
+                        namespace_blueprint_file, namespace_blueprint_id)
+                    self.manager.client.blueprints.upload(
+                        blueprint_file, self.blueprint_id)
+
+    def install(self):
+        super(AbstractRegionalCluster, self).install()
+        self._populate_status_output()
+
+    def _populate_status_output(self):
```

Comment:
It seems not in use

Reply:
2 lines up

Reply:
Do we plan to move it to capabilities ?

Reply:
let's try it out.

Reply:
This also need change in the blueprints of course, and you don't have to force refresh it like the outputs